File: Medium/739. Daily Temperatures/solution.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        
        vector<int> result(temperatures.size(), 0);
        stack<int> ST;
        
        for (int i = 0; i < temperatures.size(); i++){
            
            while (!ST.empty() && temperatures[i] > temperatures[ST.top()]){
                int day = ST.top();
                ST.pop();
                result[day] = i - day;
            }
            
            ST.push(i);
        }
        
        return result;
    }
};

File: Medium/739. Daily Temperatures/solution.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        var myStack = new Stack<int>();
        int[] result = new int[temperatures.Length];

        for (int i = 0; i < temperatures.Length; i++){
            while (myStack.Count != 0 && temperatures[i] > temperatures[myStack.Peek()]){
                int day = myStack.Pop();
                result[day] = i - day;
            }
            myStack.Push(i);
        }

        return result;
    }
}